x-pack/plugins/gis/public/shared/layers/layer.js

```diff
@@ -136,21 +139,24 @@ export class ALayer {
     return this._source.renderSourceSettingsEditor({ onChange });
   }
 
+  _findDataRequestForSource(sourceDataId) {
+    return this._dataRequests.find(dataRequest => dataRequest.getDataId() === sourceDataId);
+  }
+
+  getSourceDataRequest() {
+    return this._dataRequests.find(dataRequest => dataRequest.getDataId() === 'source');
+  }
+
   isLayerLoading() {
     return this._dataRequests.some(dataRequest => dataRequest.isLoading());
   }
 
   dataHasLoadError() {
-    return this._dataRequests.some(dataRequest => dataRequest.hasLoadError());
+    return this._descriptor && this._descriptor.errorState || false;
```

Comment:
Should `errorState` be `errorMessage`? What is the difference between the two?

Reply:
Error state is a boolean value. When an error occurs, the layer is put in `errorState` and is assigned an 'errorMessage`, assuming there is one. I could change it to something like `isInErrorState` if we wanted to be explicit on the boolean status

Reply:
I would rename to `isInErrorState`, that makes a lot of sense. Also `this._descriptor.errorState || false` should use lodash `_.get` to avoid problems like this https://github.com/elastic/kibana/pull/29275